plugins/topology/src/components/Topology/TopologySideBar/TopologyResourcesTabPanel.tsx

```diff
@@ -1,21 +1,39 @@
+import React from 'react';
 import { V1Pod, V1Service, V1ServicePort } from '@kubernetes/client-node';
 import { LongArrowAltRightIcon } from '@patternfly/react-icons';
 import { BaseNode } from '@patternfly/react-topology';
-import React from 'react';
 import ResourceName from '../../../common/components/ResourceName';
 import ResourceStatus from '../../../common/components/ResourceStatus';
 import Status from '../../../common/components/Status';
-import { IngressModel, PodModel, ServiceModel } from '../../../models';
-import { IngressData } from '../../../types/ingresses';
-import IngressRules from './IngressRules';
+import { PodModel, ServiceModel } from '../../../models';
 import TopologyResourcesTabPanelItem from './TopologyResourcesTabPaneltem';
+import RouteListSidebar from './Resources/RouteListSidebar';
+import IngressListSidebar from './Resources/IngressListSidebar';
 
 type TopologyResourcesTabPanelProps = { node: BaseNode };
 
 const TopologyResourcesTabPanel = ({
   node,
 }: TopologyResourcesTabPanelProps) => {
   const nodeData = node.getData()?.data;
+  const showIngressRoute = () => {
+    const { ingressesData, routesData } = nodeData;
+    const hasIngressData = ingressesData?.length > 0;
+    const hasRoutesData = routesData?.length > 0;
+    if (hasIngressData && hasRoutesData) {
+      return (
+        <>
+          <RouteListSidebar routesData={routesData} />
+          <IngressListSidebar ingressesData={ingressesData} />
+        </>
+      );
+    } else if (hasRoutesData && !hasIngressData) {
+      return <RouteListSidebar routesData={routesData} />;
+    } else if (hasIngressData && !hasRoutesData) {
+      return <IngressListSidebar ingressesData={ingressesData} />;
```

Comment:
```suggestion
    } 
```
We dont need this as if there is no routeData then anyways we are showing ingress-list which will show the ingress resources if available otherwise empty state.